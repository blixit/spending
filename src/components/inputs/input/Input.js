import React, { useCallback } from 'react';
import styled from 'styled-components';

const StyledInput = styled.input`
  min-height: 25px;
  lineheight: 25px;
  font-size: 16px;
  border: 0;
  border-bottom: 1px solid silver;
  background: transparent;
  &:focus{
    outline: none;
  }
`;

export const Input = (props) => {
  const { type, value, onChange, ...rest } = props;
  const change = useCallback(e => onChange(e.target.value), [onChange]);
  
  return(
    <StyledInput type={type} value={value} onChange={change} {...rest} />
  );
};

export const NumberInput = (props) => {
  const localProps = {...Object.assign({}, { type: 'number' }, props)};
  return <Input {...localProps} />;
}

export const TextInput = (props) => {
  const localProps = {...Object.assign({}, { type: 'text' }, props)};
  return <Input {...localProps} />;
}

export default Input;
