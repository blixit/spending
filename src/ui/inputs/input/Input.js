import React, { useCallback } from 'react';
import styled from 'styled-components';

const StyledInput = styled.input`
  min-height: 18px;
  lineheight: 18px;
  font-size: 0.8rem;
  border: 1px solid silver;
  border-radius: 3px;
  padding-right: 5px;
  padding-left: 5px;
  background: transparent;
  color: ${({ theme }) => theme.primary()};
  text-align: ${({ arabic }) => arabic ? 'right' : 'left'};

  &:focus{
    outline: none;
    border: 1px solid ${({ theme }) => theme.primary()};
  }

  ::-webkit-input-placeholder {
    color: ${({ theme }) => theme.primary()};
    font-size: 0.8rem;
  }
  ::-moz-placeholder {
    color: ${({ theme }) => theme.primary()};
    font-size: 0.8rem;
  }
  ::-ms-input-placeholder {
    color: ${({ theme }) => theme.primary()};
    font-size: 0.8rem;
  }
  ::placeholder {
    color: ${({ theme }) => theme.primary()};
    font-size: 0.8rem;
  }
`;

export const Input = (props) => {
  const { type, value, onChange, ...rest } = props;
  const change = useCallback(e => onChange && onChange(e.target.value), [onChange]);

  return(
    <StyledInput type={type} value={value} onChange={change} {...rest} />
  );
};

export const NumberInput = props =>
  <Input type='number' {...props} />;

export const TextInput = props =>
  <Input type='text' {...props} />;

export default Input;
