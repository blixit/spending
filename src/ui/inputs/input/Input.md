```jsx
const style = {
  flexDirection: 'column'
};

const props = {
  placeholder: 'Type your text'
};

<ExampleFlexWrap style={style}>
  <Input {...props} />
  <Input {...props} onChange={e => alert('changed')} />
</ExampleFlexWrap>
```