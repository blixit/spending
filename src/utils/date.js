/**
 * Formats a date for the backend
 * @param {Date} date the date to convert
 */
export const toBackendDate = (date) => {
  return date
    ? date.toLocaleDateString() + ' ' + date.toLocaleTimeString()
    : date;
};