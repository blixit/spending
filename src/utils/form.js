/**
* Removes the listed fields from the source
* @param {*} form The source object
* @param {*} fields The fields list
*/
export const clean = (form, fields) => {
  const result = {};
  // eslint-disable-next-line no-unused-vars
  for (const field in form) {
    if (!form.hasOwnProperty(field) || fields.includes(field)) {
      continue;
    }
    result[field] = form[field];
  }
  return result;
};
