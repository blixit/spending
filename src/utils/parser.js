export const stringify = data => {
  try {
    return JSON.stringify(data);
  } catch (error) {
    return null;
  };
}

export const parse = data => JSON.parse(data);