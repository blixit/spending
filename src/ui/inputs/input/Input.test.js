import React from 'react';

import { withTheme } from 'ui/theming/test.utils';
import render from 'testing/render';

import Input, { NumberInput, TextInput } from './Input';

describe('[inputs/input] Inputs', () => {
  describe('[inputs/input] NumberInput', () => {
    it('should render properly', () => {
      const tree = render(withTheme(NumberInput, { value: 'test' }));
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
