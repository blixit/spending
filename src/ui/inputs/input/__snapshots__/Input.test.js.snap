// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[inputs/input] Inputs [inputs/input] NumberInput should render properly 1`] = `
<input
  className="sc-bdVaJa dqOyiS"
  onChange={[Function]}
  type="number"
  value="test"
/>
`;
